tests/blocks/test_notifications.py

```diff
@@ -1,54 +1,84 @@
-
-    def test_notify_sync(self, monkeypatch):
-        async_webhook_client_mock = AsyncMock()
-        async_webhook_client_constructor_mock = MagicMock(
-            return_value=async_webhook_client_mock
-        )
-        monkeypatch.setattr(
-            "slack_sdk.webhook.async_client.AsyncWebhookClient",
-            async_webhook_client_constructor_mock,
-        )
-
-        url = "http://example.com/slack"
-
-        @flow
-        def test_flow():
-            block = SlackWebhook(url=url)
-            block.notify("test")
-
-        test_flow()
-        async_webhook_client_constructor_mock.assert_called_once_with(url=url)
-        async_webhook_client_mock.send.assert_called_once_with(text="test")
-
-    def test_is_picklable(self):
-        block = SlackWebhook(url="http://example.com/slack")
+def reload_modules():
+    """
+    Reloads the prefect.blocks.notifications module so patches to modules it imports
+    will be visible to the blocks under test.
+    """
+    try:
+        reload(prefect.blocks.notifications)
+    except UserWarning as ex:
+        # ignore the warning Prefect gives when reloading the notifications module
+        # because we reload prefect itself immediately afterward.
+        pass
+
+    reload(prefect)
+
+
+# A list of the notification classes Pytest should use as parameters to each method in TestAppriseNotificationBlock
+notification_classes = sorted(
+    AppriseNotificationBlock.__subclasses__(), key=lambda cls: cls.__name__
+)
+
+
+@pytest.mark.parametrize("block_class", notification_classes)
+class TestAppriseNotificationBlock:
+    """
+    Checks for behavior expected from Apprise-based notification blocks.
+    """
+
+    async def test_notify_async(self, block_class: Type[AppriseNotificationBlock]):
+        with patch("apprise.Apprise", autospec=True) as AppriseMock:
+            reload_modules()
+
+            apprise_instance_mock = AppriseMock.return_value
+            apprise_instance_mock.async_notify = AsyncMock()
+
+            block = block_class(url="https://example.com/notification")
+            await block.notify("test")
+
+            AppriseMock.assert_called_once()
+            apprise_instance_mock.add.assert_called_once_with(
+                block.url.get_secret_value()
+            )
+            apprise_instance_mock.async_notify.assert_awaited_once_with(
+                body="test", title=None, notify_type=PrefectNotifyType.DEFAULT
+            )
+
+    def test_notify_sync(self, block_class: Type[AppriseNotificationBlock]):
+        with patch("apprise.Apprise", autospec=True) as AppriseMock:
+            reload_modules()
+
+            apprise_instance_mock = AppriseMock.return_value
+            apprise_instance_mock.async_notify = AsyncMock()
+
+            url = "https://example.com/notification"
+
+            @prefect.flow
+            def test_flow():
+                block = block_class(url=url)
+                block.notify("test")
```

Comment:
This flow can also be removed.

Reply:
Makes sense. It is gone now. I had left that one because it was in the original `test_notify_sync`, but I suppose we're calling the block synchronously whether we do it in a flow or not.